dev/reference/tools: update sync-diff-inspector.md (#1216)
pingcap/docs: dev/reference/tools/sync-diff-inspector.md

> - TiDB uses the `utf8_bin` collation. If you need to compare the data in MySQL with that in TiDB, pay attention to the collation configuration of MySQL tables. If the primary key or unique key is the `varchar` type and the collation configuration in MySQL differs from that in TiDB, then the final check result might be incorrect because of the collation issue. You need to add collation to the sync-diff-inspector configuration file.
> - If you configure the `tidb-instance-id` to divide data into chunks according to TiDB statistics, try to guarantee the accuracy of statistics. You can manually run the `analyze table {table_name}` command when the TiDB server *workload is light*.
> - Pay special attention to `table-rules`. If you configure `schema-pattern="test1"` and `target-schema="test2"`, the `test1` schema in the source database and the `test2` schema in the target database are compared. If the source database has a `test2` schema, this `test2` schema is also compared with the `test2` schema in the target database.

> > - sync_diff_inspector saves the check status to  `sync_diff_inspector` schema in the target database during checking. You need to ensure that database users have read and write permissions to the corresponding schemas.

Reviewer: - Remove an extra typo

```suggestion
> - sync_diff_inspector saves the check status to the `sync_diff_inspector` schema in the target database during checking. You need to ensure that database users have read and write privileges of corresponding schemas.
```

# If disabled, data is compared line by line.
use-checksum = true

> # If set to true, data is checked by calculating checksum. Data will not be checked after data is detected, even if the upstream and downstream checksums are inconsistent.

Reviewer: Strictly speaking, the subject of "If set to true" and "data is ..." must be the same.

```suggestion
# If it is set to true, data is checked only by calculating checksum. Data will not be checked after inspection, even if the upstream and downstream checksums are inconsistent.
```

only-use-checksum = false

> # Whether to use the checkpoint of the last check. If enabled, only check the last unverified chunks and chunks that failed the verification.

Reviewer: ```suggestion
# Whether to use the checkpoint of the last check. If it is enabled, the inspector only checks the last unverified chunks and chunks that failed the verification.
```

use-checkpoint = true

> # If set to true, data check is ignored. 

Reviewer: ```suggestion
# If it is set to true, data check is ignored. 
```

> # If set to false, data is checked.

Reviewer: ```suggestion
# If it is set to false, data is checked.
```

ignore-data-check = false

> # If set to true, the table struct comparison is ignored. 

Reviewer: ```suggestion
# If it is set to true, the table struct comparison is ignored. 
```
